File: ios/EscalaMinisterial/Models/Escala.swift
import Foundation

struct Escala: Codable, Identifiable {
    var id: Int?
    var eventoId: Int?
    var eventoNome: String?
    var eventoData: String?
    var eventoHorario: String?
    var dataAtribuicao: String?
    var observacao: String?
    var status: String
    var escalaMinistros: [EscalaMinistro]

    enum StatusEscala: String, CaseIterable {
        case proposta   = "PROPOSTA"
        case aprovada   = "APROVADA"
        case confirmada = "CONFIRMADA"
        case cancelada  = "CANCELADA"

        var label: String {
            switch self {
            case .proposta:   return "Proposta"
            case .aprovada:   return "Aprovada"
            case .confirmada: return "Confirmada"
            case .cancelada:  return "Cancelada"
            }
        }
    }

    var statusEnum: StatusEscala? { StatusEscala(rawValue: status) }

    var eventoDataFormatada: String? {
        guard let d = eventoData, let date = DateUtils.parseDate(d) else { return eventoData }
        return DateUtils.formatDate(date)
    }
}

struct EscalaMinistro: Codable, Identifiable {
    var id: Int?
    var ministroId: Int
    var ministroNome: String?
    var ministroFuncao: String?
    var confirmacaoMinistro: Bool
    var substituido: Bool
}

File: android/feature/escalas/src/main/java/com/escala/ministerial/feature/escalas/domain/model/Escala.kt
package com.escala.ministerial.feature.escalas.domain.model

import java.time.LocalDate

data class Escala(
    val id: Long,
    val eventoId: Long,
    val eventoNome: String,
    val eventoData: LocalDate?,
    val eventoHorario: String?,
    val dataAtribuicao: LocalDate?,
    val observacao: String?,
    val status: StatusEscala,
    val ministros: List<EscalaMinistro>,
    val totalMinistros: Int = ministros.size,
)

data class EscalaMinistro(
    val id: Long,
    val ministroId: Long,
    val ministroNome: String,
    val confirmacaoMinistro: Boolean,
    val substituido: Boolean,
)

enum class StatusEscala(val label: String) {
    PROPOSTA("Proposta"),
    APROVADA("Aprovada"),
    CONFIRMADA("Confirmada"),
    CANCELADA("Cancelada"),
}
